internal/proxy/multi_rate_limiter.go

```diff
@@ -49,80 +50,98 @@ func GetQuotaErrorString(errCode commonpb.ErrorCode) string {
 // MultiRateLimiter includes multilevel rate limiters, such as global rateLimiter,
 // collection level rateLimiter and so on. It also implements Limiter interface.
 type MultiRateLimiter struct {
-	globalRateLimiter *rateLimiter
 	// TODO: add collection level rateLimiter
-	quotaStatesMu sync.RWMutex
-	quotaStates   map[milvuspb.QuotaState]commonpb.ErrorCode
+	quotaStatesMu      sync.RWMutex
+	collectionLimiters map[int64]*rateLimiter
 }
 
 // NewMultiRateLimiter returns a new MultiRateLimiter.
 func NewMultiRateLimiter() *MultiRateLimiter {
-	m := &MultiRateLimiter{}
-	m.globalRateLimiter = newRateLimiter()
+	m := &MultiRateLimiter{
+		collectionLimiters: make(map[int64]*rateLimiter, 0),
+	}
 	return m
 }
 
 // Check checks if request would be limited or denied.
-func (m *MultiRateLimiter) Check(rt internalpb.RateType, n int) commonpb.ErrorCode {
+func (m *MultiRateLimiter) Check(collectionID int64, rt internalpb.RateType, n int) commonpb.ErrorCode {
```

Comment:
Returning the error redefined by merr is better.

Reply:
for now, the error code here are expose to sdk side by response, it's hard to warp it, may cause side effect.